### PR TITLE
Plain Action Refactor

### LINKED ISSUE #40: Download the latest version of the zarf binary by default
Currently, this action requires you to specify what version of zarf you want to install/use.

This feels unnecessary, as most users probably will not be too concerned about what version they are using. Most users will likely be happy with using the latest version.

Desired outcomes:
- users are not required to specify what version of zarf they want (make the"version" input variable optional)
- the action downloads the latest version of zarf if no version is supplied/specified

### PR DESCRIPTION
**Pull request by jeff-mccoy**
This PR changes the overall strategy for the Zarf action. We want to keep maximum portability for Zarf using in CI so features will be rolled into the Zarf CLI vs a complex Github action. The primary goal of this action will be to make the adoption of Zarf in a workflow frictionless, lightweight and simple. This will be achieved by focusing on simply downloading the Zarf release artifacts as the main purpose of the action.

Closes #40 
Related to #41

### CODE REVIEW

**Comment by lucasrod16** [comment]
I like the overall approach and rewrite. It simplifies things greatly, makes it more portable, and offloads any heavy business logic/functionality to Zarf.

File: action.yml [hunk 1]
```diff
@@ -1,12 +1,71 @@
...
+description: "Installs Zarf and includes the Zarf CLI in the PATH"
 inputs:
   version:
-    description: 'The Version of Zarf to Install'
+    description: "The version of Zarf to install. Use v0.22.2 for example."
     required: true
   download-init-package:
-    description: 'Download Zarf Init Package'
+    description: "Download the corresponding Zarf init package."
     required: true
```
[the remark this answers is not in the document]
Comment (jeff-mccoy) [reply action.yml:9]: Oh good catch, yes that was a mistake.  It should have defaulted to false and been optional, will fix that in a min.

File: action.yml [hunk 2]
```diff
@@ -1,12 +1,71 @@
-name: 'Setup Zarf'
-description: 'GitHub Action for Installing Zarf'
+name: "Setup Zarf"
+description: "Installs Zarf and includes the Zarf CLI in the PATH"
 inputs:
   version:
-    description: 'The Version of Zarf to Install'
+    description: "The version of Zarf to install. Use v0.22.2 for example."
     required: true
```
[the remark this answers is not in the document]
Comment (jeff-mccoy) [reply action.yml:6]: I actually started with that and then dropped it to match your existing schema. There's a couple things we can think about:

- getting latest release seems very reasonable
- we may start cutting Zarf pre-releases so we would need to decide if this action should support using those in more experimental envs.  Thoughts?

[the remark this answers is not in the document]
Comment (lucasrod16) [reply action.yml:6]: Getting the latest release by default sounds good to me. If we leave the version input variable and make it an optional input, then users can specify a pre-release version if desired as well

[the remark this answers is not in the document]
Comment (jeff-mccoy) [reply action.yml:6]: So looking at this, the right way to do it is using the Github API, but they rate-limit that so we'd need to use a GH token like this doc says: https://docs.github.com/en/rest/guides/getting-started-with-the-rest-api#authentication-example-for-github-actions.

You can get that latest release via `https://api.github.com/repos/defenseunicorns/zarf/releases/latest` or `gh api repos/defenseunicorns/zarf/releases`.

[the remark this answers is not in the document]
Comment (lucasrod16) [reply action.yml:6]: Yeah I think either one would work well for us

[the remark this answers is not in the document]
Comment (jeff-mccoy) [reply action.yml:6]: issue will be we won't have access to the token without asking for it, so I think that's a messy solution too. Hmmm....

[the remark this answers is not in the document]
Comment (jeff-mccoy) [reply action.yml:6]: How about we keep it as-is on this PR and use #40 to work on another PR to address this?

File: action.yml [hunk 3]
```diff
@@ -1,12 +1,78 @@
...
+        # Set the architecture variable
+        case ${{ runner.arch }} in
+          X64)
+            arch="amd64"
+            ;;
+          ARM64)
+            arch="arm64"
+            ;;
+          *)
+            echo "Unsupported architecture X64"
```
Comment (lucasrod16) [inline action.yml:38]: Are we wanting to include the X64 in this `echo` command?

File: README.md [hunk 4]
```diff
@@ -1,68 +1,88 @@
...
-
-This GitHub action is considered experimental because of its early-stage development.
+This action will optionally download a Zarf init package. To learn more about Zarf init packages and their use-cases, see the [Zarf docs](https://docs.zarf.dev/docs/user-guide/zarf-packages/the-zarf-init-package).
 
 ## Usage
 
 ```yaml
 uses: defenseunicorns/setup-zarf@main
 with:
   version: <version> # Required
```
Comment (lucasrod16) [inline README.md:12]: Since the version input is optional now, we can remove the # Required comment

File: README.md [hunk 5]
```diff
@@ -1,68 +1,88 @@
...
+#
 
-### Package Scripts
+## Inputs
 
-When changes are made to the javascript source code, run `npm run all` to execute `eslint` against the code and recompile the code into the `dist/index.js` file.
+### version
 
-&nbsp;
+- Required
```
Comment (lucasrod16) [inline README.md:81]: Think we just need to update this to Optional

**Review by lucasrod16 — APPROVED** [review]

**Pull request merged**